patches/server/0978-add-PlayerCreateMapEvent.patch

```diff
@@ -0,0 +1,41 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chmodsayshello <chmodsayshello@hotmail.com>
+Date: Tue, 2 May 2023 18:22:35 +0200
+Subject: [PATCH] add PlayerCreateMapEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/item/EmptyMapItem.java b/src/main/java/net/minecraft/world/item/EmptyMapItem.java
+index e92bb09dd3218d5a13e6251bddd0812acadda2be..8b07c86f22299e9c89e70603abf77ef78209ac31 100644
+--- a/src/main/java/net/minecraft/world/item/EmptyMapItem.java
++++ b/src/main/java/net/minecraft/world/item/EmptyMapItem.java
+@@ -6,6 +6,7 @@ import net.minecraft.world.InteractionHand;
+ import net.minecraft.world.InteractionResultHolder;
+ import net.minecraft.world.entity.player.Player;
+ import net.minecraft.world.level.Level;
++import io.papermc.paper.event.player.PlayerCreateMapEvent; //Paper
+ 
+ public class EmptyMapItem extends ComplexItem {
+     public EmptyMapItem(Item.Properties settings) {
+@@ -18,13 +19,18 @@ public class EmptyMapItem extends ComplexItem {
+         if (world.isClientSide) {
+             return InteractionResultHolder.success(itemStack);
+         } else {
+-            if (!user.getAbilities().instabuild) {
+-                itemStack.shrink(1);
+-            }
+-
+             user.awardStat(Stats.ITEM_USED.get(this));
+             user.level.playSound((Player)null, user, SoundEvents.UI_CARTOGRAPHY_TABLE_TAKE_RESULT, user.getSoundSource(), 1.0F, 1.0F);
+             ItemStack itemStack2 = MapItem.create(world, user.getBlockX(), user.getBlockZ(), (byte)0, true, false);
++            //Paper start
++            PlayerCreateMapEvent playerCreateMapEvent = new PlayerCreateMapEvent((org.bukkit.entity.Player) user.getBukkitEntity(), itemStack2.asBukkitCopy());
++            world.getCraftServer().getPluginManager().callEvent(playerCreateMapEvent);
++            if(playerCreateMapEvent.isCancelled()){
++                return InteractionResultHolder.consume(itemStack);
++            }else if (!user.getAbilities().instabuild) { //moved down here so an item gets removed only if the event wasn't cancelled!
++                itemStack.shrink(1);
++            }
++            //Paper end
```

Comment:
Fix spacing after comments and brackets and before brackets

Reply:
Should be better now, but for time being I'll leave this open